CMakeLists.txt

```diff
@@ -13,7 +13,6 @@ set (CMAKE CXX_FLAGS "-Wall -Wextra -pedantic -Wno-unused-parameter -faligned-ne
 add_definitions(-Wfatal-errors)
 add_definitions(-Wno-psabi)
 add_definitions(-DBOOST_LOG_DYN_LINK)
-add_definitions(-mfpu=neon-fp-armv8 -ftree-vectorize)
```

Comment:
I'd quite like to like these compile flags in place for platforms that support them. Do you know of a way to do that? Thanks!

Reply:
Sorry, no easy way (and the buildroot view/preferred way is to avoid package specific  heuristics to determine architecture specific compile flags - in buildroot the preferred way is build all packages specific to your selected target architecture/cpu etc. by a global configure/selection).

One problem is (for cross compiling) the detection of the target arch and/or the suitable/supported compiler flags (and not all cross-compiler available flags/features will be available on the target platform, specially for multi-arch/targets compilers).

The next/other problem is that gcc compiler distinguish between used/maximum feature set and optimization for target arch...

Do not know a perfect example but a quick search found e.g. [1] (did not check if it works for cross-compiling)...

But a simple solution is to implement one/some cmake option(s) to enable target specific flags/optimizations (easy to not to use and/or override with global/package specific compiler flags)...

[1] https://github.com/seredat/karbowanec/blob/master/CMakeLists.txt

Reply:
Would it work for you to have a TARGET variable which can default to (say) "armv8" but such that you can invoke cmake with "-DTARGET=other" or something like that?

Reply:
Yes, should definitely work (would name it a little more specific like ENABLE_COMPILE_FLAGS_FOR_TARGET_ARCH?)...

Reply:
Would this do the trick for you?
```
IF (NOT ENABLE_COMPILE_FLAGS_FOR_TARGET)
  set(ENABLE_COMPILE_FLAGS_FOR_TARGET "armv8")
endif()
if ("${ENABLE_COMPILE_FLAGS_FOR_TARGET}" STREQUAL "armv8")
  add_definitions(-mfpu=neon-fp-armv8 -ftree-vectorize)
endif()
```
If you were ok to put the changes we've discussed into your branch then I'd be happy to test and merge if it all works fine for our "default Pi user". Thanks!

Reply:
Yes, works for the buildroot case, merge request updated...